### DISCUSSION TITLE
To place and draw components at specific coordinates using Java with Structurizr

### DISCUSSION THREAD

**Discussion by ghost** [structurizr/java#269] [Q&A] [answered]
I am using Structurizr java for our enterprise application. For placing or drawing an element at specific co-ordinates, I have tried using ElementView with AutomaticLayout and without AutomaticLayout. But its not helping to place an element at a position. Im following this way:

ElementView eleView = deploymentView.getElements().stream()
				.filter(ev -> ev.getElement().getName().equals("nodeName")).findFirst().orElse(null);			

if (eleView != null) {
	eleView.setX(30);
	eleView.setY(40);
}

Please suggest me a way to order the elements through java. Also please suggest me if im doing this in a wrong way.

**Comment by ghost** [comment]
Hi,

Can anyone please help me here. I would really need this in our project for placing the elements.

**Comment by simonbrowndotje** [comment, marked as answer]
This code will work - if you're pushing your workspace to an on-premises installation or the cloud service, be sure to call `setMergeFromRemote(false)` on your [WorkspaceApiClient](https://github.com/structurizr/java/blob/master/structurizr-client/src/main/java/com/structurizr/api/WorkspaceApiClient.java#L149) instance to force push your layout information.

**Reply by ghost** [reply]
Thank you so much for your assistance. I'm able to set coordinates. This helped me a lot.
